### PR TITLE
[SYCL][ESIMD] Reenable the tests

### PR DESCRIPTION
**Pull request by fineg74**

### CODE REVIEW

File: sycl/test-e2e/ESIMD/matrix_transpose.cpp [hunk 1]
```diff
@@ -5,11 +5,9 @@
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 //
 //===----------------------------------------------------------------------===//
-// FIXME: Investigate Windows-specific failures
-// REQUIRES: TEMPORARY_DISABLED
-// UNSUPPORTED: cuda || hip
+// REQUIRES: gpu
+// UNSUPPORTED: cuda || hip || esimd_emulator
```
Comment (v-klochkov) [inline sycl/test-e2e/ESIMD/matrix_transpose.cpp:9]: Ok, but I see you tried to fix the issue on esimd_emulator, and may know why it fails there.
If so, can you please add TODO: comment here explainings why test fails on emulator?

File: sycl/test-e2e/ESIMD/matrix_transpose.cpp [hunk 2]
```diff
@@ -5,11 +5,9 @@
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 //
 //===----------------------------------------------------------------------===//
-// FIXME: Investigate Windows-specific failures
```
Comment (bader) [inline sycl/test-e2e/ESIMD/matrix_transpose.cpp:8]: Could you keep it disabled on Windows until we enable the validation on Windows, please?
@uditagarwal97, FYI.

File: sycl/test-e2e/ESIMD/matrix_transpose.cpp (hunk 1 above)
Comment (v-klochkov) [reply sycl/test-e2e/ESIMD/matrix_transpose.cpp:9]: @fineg74 - please address this comment. If you know why emulator fails, please keep a comment with your opinion explaining why it is marked as unsupported.

File: sycl/test-e2e/ESIMD/matrix_transpose.cpp [hunk 3]
```diff
@@ -5,10 +5,10 @@
...
-// REQUIRES: TEMPORARY_DISABLED
-// TODO: esimd_emulator fails due to unimplemented __esimd_oword_ld_unaligned
-// XFAIL: esimd_emulator
+// REQUIRES: gpu
```
Comment (v-klochkov) [inline sycl/test-e2e/ESIMD/matrix_transpose.cpp:8]: Please re-base your patch once again.
The current/head version of this test (https://github.com/intel/llvm/blob/sycl/sycl/test-e2e/ESIMD/matrix_transpose.cpp) doesn't need this "REQUIRES: gpu" line
Same for cuda || hip.
Same for gpu-intel-gen9 && windows

**Review by v-klochkov — APPROVED** [review]

**Pull request merged**